### ISSUE TITLE
Deadlock trying to upgrade to Vuforia 7.2 

### ISSUE THREAD

**Issue by setoelkahfi** [jiulongw/swift-unity#89]
Anybody ever try to upgrade to Vuforia 7.2 with this method? Please share your finding, I'm facing a deadlock 😁

**Comment by melvinrudolph** [comment]
Could you post your solution? Would be so awesome !!

**Comment by setoelkahfi** [comment]
Sure. Maybe a PR for this repository so we can discuss the best practice.

**Comment by melvinrudolph** [comment]
Okay so most of the errors are gone now but it's still not entirely working.
What im left with now is:

Undefined symbols for architecture arm64:
  "Vuforia::TextTracker::getClassType()", referenced from:
      _textTrackerSetRegionOfInterest in libVuforiaWrapper.a(TextTracker.o)
      _textTrackerGetRegionOfInterest in libVuforiaWrapper.a(TextTracker.o)
      _trackerManagerInitTracker in libVuforiaWrapper.a(TrackerManager.o)
      _trackerManagerDeinitTracker in libVuforiaWrapper.a(TrackerManager.o)
      _wordListLoadWordList in libVuforiaWrapper.a(WordList.o)
      _wordListAddWordsFromFile in libVuforiaWrapper.a(WordList.o)
      _wordListAddWordU in libVuforiaWrapper.a(WordList.o)
      ...
  "Vuforia::Word::getClassType()", referenced from:
      TypeMapping::getType(int) in libVuforiaWrapper.a(TypeMapping.o)
      TypeMapping::getTypeID(Vuforia::Type) in libVuforiaWrapper.a(TypeMapping.o)
  "Vuforia::WordResult::getClassType()", referenced from:
      _updateQCAR in libVuforiaWrapper.a(Tracker.o)
      _determineNewWords in libVuforiaWrapper.a(Tracker.o)
      _wordGetLetterMask in libVuforiaWrapper.a(Tracker.o)
      _wordGetLetterBoundingBoxes in libVuforiaWrapper.a(Tracker.o)
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

Do you have any idea how to fix that ? Couldn't find any working solution for this arm64 error

**Comment by setoelkahfi** [comment]
That's something I encountered before moving the Vuforia.framework first to my xcode build folder.

Maybe try to clean, restart Xcode, restart your Mac? You know, Xcode :D

**Comment by melvinrudolph** [comment]
Yeah i also had about 200 of these error messages before i tried your solution. These are all gone now, thanks to you, except the ones i mentioned :D maybe i'll find a way

**Comment by ericqqqqq** [comment]
Hi setoelkahfi,
thank you for uploading your sample. I'm testing your demo, however, I get an error from DeviceSetting.mm. 
at line255 of DeviceSetting.mm, I'm receiving an error: control may reach end of non-void function due to nothing is returned for endif.

I tried to return deviceUnknown; However, I can't modify the file, since every time I build my project, this file is automatically modified by other application.

How should I fix this problem?
Thanks

**Comment by ericqqqqq** [comment]
Hi setoelkahfi,
Thanks to your demo, I successfully built my project, however, I get errors in run time and fails at launch screen

the error is attached.
<img width="1438" alt="screen shot 2018-08-01 at 4 12 19 pm" src="https://user-images.githubusercontent.com/41719259/43546174-f31568b4-95a5-11e8-9a56-3ea196e7b43a.png">

My unity is: UNITY_RUNTIME_VERSION = 2018.2.1f1;
Xcode is: Version 9.4.1 (9F2000)

**Comment by setoelkahfi** [comment]
Oh, I forgot my PR that should fix the issue.

@ericqqqqq that's because Unity detect there's a launcher storyboard (whish is in your project), but failed to load its launcher (because we don't copy it to our project)

Quick solution is rename your launcher storyboard to something else. Meanwhile I'll update my PR.

**Comment by ericqqqqq** [comment]
@setoelkahfi  
Thank you very much! 
So I need to rename my own storyboard and in the setting, set the main interface to the renamed storyboard. Is that right?

**Comment by ericqqqqq** [comment]
@setoelkahfi  thank you very much! I got it!

**Comment by capnmidnight** [comment]
Would it be possible to get these instructions added to the README?

**Comment by setoelkahfi** [comment]
@capnmidnight I think it's not a problem. You can make the pr based on this.
